src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -195,90 +113,129 @@ const AddressVerificationFlow: React.FC<AddressVerificationFlowProps> = ({
       ...(inputAddress as any),
       key: USER_INPUT,
       recommended: false,
     }
-
-    if (hasError) {
-      const fallbackOption = fallbackFromFormValues(verificationInput)
-      const verifiedBy = AddressVerifiedBy.USER
-      onChosenAddress(verifiedBy, fallbackOption.address)
+    if (verificationStatus === "VERIFIED_NO_CHANGE") {
+      verificationPath = VerificationPath.IMMEDIATE_CONFIRM
+      addressOptions = [inputOption]
     } else {
-      if (verificationStatus === "VERIFIED_NO_CHANGE") {
-        const verifiedBy = AddressVerifiedBy.ARTSY
-        onChosenAddress(verifiedBy, inputOption.address)
+      if (verificationStatus === "VERIFIED_WITH_CHANGES") {
+        verificationPath = VerificationPath.SUGGESTIONS
+
+        const suggestedOptions = (suggestedAddresses || [])
+          .slice(0, 1)
+          .map(({ address, lines }: any, index) => ({
+            key: `suggestedAddress-${index}`,
+            recommended: index === 0,
+            lines: lines,
+            address: address,
+          }))
+        addressOptions = [...suggestedOptions, inputOption]
       } else {
-        if (verificationStatus === "VERIFIED_WITH_CHANGES") {
-          setModalType(ModalType.SUGGESTIONS)
-          trackEvent({
-            action_type: "validationAddressViewed",
-            context_module: ContextModule.ordersShipping,
-            context_page_owner_type: OwnerType.ordersShipping,
-            context_page_owner_id: contextPageOwnerSlug,
-            user_id: userId,
-            flow: "user adding shipping address",
-            subject: "Confirm your delivery address",
-            option: "suggestions",
-          })
-
-          const suggestedOptions = (suggestedAddresses || [])
-            .slice(0, 1)
-            .map(({ address, lines }: any, index) => ({
-              key: `suggestedAddress-${index}`,
-              recommended: index === 0,
-              lines: lines,
-              address: address,
-            }))
-          setAddressOptions([...suggestedOptions, inputOption])
-        } else {
-          setAddressOptions([inputOption])
-          setModalType(ModalType.REVIEW_AND_CONFIRM)
-          trackEvent({
-            action_type: "validationAddressViewed",
-            context_module: ContextModule.ordersShipping,
-            context_page_owner_type: OwnerType.ordersShipping,
-            context_page_owner_id: contextPageOwnerSlug,
-            user_id: userId,
-            flow: "user adding shipping address",
-            subject: "Check your delivery address",
-            option: "review and confirm",
-          })
-        }
+        addressOptions = [inputOption]
+        verificationPath = VerificationPath.REVIEW_AND_CONFIRM
       }
     }
+  }
+
+  const modalTitle = modalTitles[verificationPath]
+
+  const [selectedAddressKey, setSelectedAddressKey] = useState<
+    AddressOptionKey | undefined
+  >(addressOptions[0]?.key)
+
+  // perform only once on mount
```

Comment:
This comment isn't actually accurate, since there are dependencies in the `useEffect` array. If you want it to only perform once, need to explicitly use `[]`

Reply:
Ah, i guess in this case i don't need the ref - i thought there was a risk of double-firing events without it and was hesitant to disable the exhaustive deps rule but i can do it if you say its ok.

Reply:
If you want something to fire once, the effect fires on mount -- and if there are no deps to track, it will never fire again. The linter doesn't know your intent, so it will always warn.

Reply:
Updated to simply fire once on mount with `[]`.